scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,222 @@
+    image: The image to use (defaults to quay.io/kubevirt/must-gather)
+    api-key: Use a generated API key instead of a username and login
+
+Requirements:
+
+    Openshift 4.1
+    Python 3.6
+    A Bugzilla account for www.bugzilla.redhat.com
+
+This script attaches the result of the must-gather command, as executed
+by the kubevirt must-gather image, to the supplied bugzilla id on the
+Red Hat bugzilla website.
+It first creates an output subdirectory in the working directory named
+gather-files/ and then executes the following command:
+'oc adm must-gather --image=quay.io/kubevirt/must-gather
+--dest-dir=gather-files/' and pipes the output to
+gather-files/must-gather.log
+In order to meet the maximum attachment file sizes, it searches the
+created directory for any log files that exceed 100,000 lines, and if
+there are any, it trims those files down to 100,000 lines.
+It then creates a time-stamped archive file to compress the attachment
+and prepare it for upload.
+After doing so, the attachment is encoded as a base64 string.
+In order to authenticate against the Bugzilla website, a username and
+password are prompted. A POST request is then sent to the Bugzilla
+website. If there are any errors (invalid ID or invalid login), the
+script prompts for those and retries the request until there are no
+errors.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import tarfile
+import datetime
+import base64
+from getpass import getpass
+import requests
+
+MAX_LOGLINES = 100000
+
+BUGZILLA_URL = "https://www.bugzilla.redhat.com"
+
+HEADERS = {'Content-type': 'application/json'}
+
+LOGFOLDER = "gather-files/"
+
+OUTPUT_FILE = "must-gather.log"
+
+ARCHIVE_NAME = "must-gather"
+
+IMAGE = "quay.io/kubevirt/must-gather"
+
+def main():
+    """Main function"""
+
+    # Start with getting command-line argument(s)
+    parser = argparse.ArgumentParser(description="Sends the result of must-gather to Bugzilla.")
+    parser.add_argument("ID", metavar="id", type=int,
+                        help="The ID of the bug in Bugzilla")
+    parser.add_argument("--image", metavar="image",
+                        help="The image to use for must-gather")
+    parser.add_argument("--api-key", metavar="api-key",
+                        help="Optional API key instead of username and login (will disable prompts to retry)")
+    args = parser.parse_args()
+
+    bug_id = args.ID
+
+    if args.image:
+        image = args.image
+    else:
+        image = IMAGE
+
+    use_api_key = args.api_key != None
+
+    # If the log folder already exists, delete it
+    if os.path.isdir(LOGFOLDER):
+        shutil.rmtree(LOGFOLDER)
+
+    # Make a new log folder
+    os.mkdir(LOGFOLDER)
+
+    # Open the output file
+    with open(LOGFOLDER + OUTPUT_FILE, "w+") as out_file:
+        # Run oc adm must-gather with the appropriate image and dest-dir
+        print("Running must-gather")
+        subprocess.run(
```

Comment:
Please check the return code and print stderr if any. There is no need to continue if the command failed.

Reply:
Done in recent commit